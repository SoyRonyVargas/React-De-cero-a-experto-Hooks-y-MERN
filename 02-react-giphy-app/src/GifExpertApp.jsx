import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({ def = [] }) => {
    
    const [categories, setCategories] = useState(def);

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories } />
            <hr />

            <ol>
                {
                    categories.map( category  => (
                        <GifGrid 
                            key={ category }
                            category={ category }
                        />
                    ))
                }
            </ol>

        </>
    )
}
