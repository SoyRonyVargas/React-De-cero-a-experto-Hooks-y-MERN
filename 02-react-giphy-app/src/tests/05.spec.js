import { GifExpertApp } from '../GifExpertApp'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import React from 'react'

describe('Pruebas al componente GifExpertApp', () => {
    
    let wrapper = shallow(<GifExpertApp/>)

    test('Match con snapshot', () => {

        expect(wrapper).toMatchSnapshot()
        
    })
    
    test('Debe de mostrar 2 hijos ', () => {

        const categories = ['one punch', 'goku']
        
        wrapper = shallow(<GifExpertApp def={categories} />)

        expect(wrapper).toMatchSnapshot()

        expect(wrapper.find('GifGrid').length).toBe(categories.length)
        
    })
    

})
