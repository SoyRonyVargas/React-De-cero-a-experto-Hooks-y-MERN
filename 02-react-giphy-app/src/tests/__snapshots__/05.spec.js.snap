// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente GifExpertApp Debe de mostrar 2 hijos  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="one punch"
      key="one punch"
    />
    <GifGrid
      category="goku"
      key="goku"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas al componente GifExpertApp Match con snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
